consensus/progpow/consensus.go

```diff
@@ -660,38 +661,28 @@ func (progpow *Progpow) Finalize(chain consensus.ChainHeaderReader, batch ethdb.
 	var multiSet *multiset.MultiSet
 	if chain.IsGenesisHash(header.ParentHash(nodeCtx)) {
 		multiSet = multiset.New()
-		alloc := core.ReadGenesisAlloc("genallocs/gen_alloc_quai_"+nodeLocation.Name()+".json", progpow.logger)
-		progpow.logger.WithField("alloc", len(alloc)).Info("Allocating genesis accounts")
+		alloc, err := genallocs.AllocateGenesisAccounts("genallocs/gen_alloc_quai_" + nodeLocation.Name() + ".json")
```

Comment:
why are we only doing this if parent is genesis? i dont see any other place where we call the add locked balance method